model/train.py

```diff
@@ -588,99 +599,122 @@ def encode_tokens(labels):
-          summary = jax.tree_map(lambda x: x / denominator, metrics_sums)  # pylint: disable=cell-var-from-loop
-          summary["sentence_accuracy"] = sentence_acc / sentence_denominator
-          summary["learning_rate"] = lr
-          summary = {"train_" + k: v for k, v in summary.items()}
-          writer.write_scalars(step, summary)
-          train_metrics = []
-
-        with report_progress.timed("eval_train"):
-          eval_results_training = evaluate(
-              p_eval_step=p_eval_step,
-              target=optimizer.target,
-              eval_ds=eval_train_ds,
-              num_eval_steps=config.num_eval_train_steps)
-          writer.write_scalars(
-              step, {"eval_train_" + k: v for k, v in eval_results_training.items()})
-              
-        with report_progress.timed("predictions_train"):
-          exemplars, train_acc = decode_and_calculate_acc(
-              p_pred_step=p_pred_step,
-              p_init_cache=p_init_cache,
-              target=optimizer.target,
-              config=config,
-              predict_ds=predict_train_ds,
-              decode_tokens=decode_tokens,
-              encode_tokens=encode_tokens,
-              max_predict_length=config.max_predict_length,
-              max_predict_loops=1)
-          writer.write_scalars(step, {"pred_train_acc": train_acc})
-          writer.write_texts(step, {"samples": exemplars})
-           
-        with report_progress.timed("eval"):  
-          eval_results = evaluate(
-              p_eval_step=p_eval_step,
-              target=optimizer.target,
-              eval_ds=eval_ds,
-              num_eval_steps=config.num_eval_steps)
-          writer.write_scalars(
-              step, {"eval_" + k: v for k, v in eval_results.items()})
-
-        with report_progress.timed("predictions"):
-          exemplars, test_acc = decode_and_calculate_acc(
-              p_pred_step=p_pred_step,
-              p_init_cache=p_init_cache,
-              target=optimizer.target,
-              config=config,
-              predict_ds=predict_ds,
-              decode_tokens=decode_tokens,
-              encode_tokens=encode_tokens,
-              max_predict_length=config.max_predict_length,
-              max_predict_loops=config.num_predict_loops)
-          writer.write_scalars(step, {"pred_test_accuracy": test_acc})
-          writer.write_texts(step, {"samples": exemplars})
-
-      # Save a checkpoint on one host after every checkpoint_freq steps.
-      save_checkpoint = (step % config.checkpoint_every_steps == 0 or
-                         is_last_step)
-      if config.save_checkpoints and save_checkpoint and jax.process_index() == 0:
-        with report_progress.timed("checkpoint"):
-          checkpoints.save_checkpoint(workdir, jax_utils.unreplicate(optimizer),
-                                      step)
+  if config.just_do_pred:
+      exemplars, test_acc, predicted_list = decode_and_calculate_acc(
```

Comment:
Maybe it's the Github visualization, but there some missing indentation here, no?

Reply:
It looks like it was the Github visualization!